### PR TITLE
[python-updates] 2024-04-18 - minor updates

### PR DESCRIPTION
**Pull request by mweinelt**
This is a bulk update of python packages, that is restricted to minor version bumps.

### Known issues
- First packages requiring `numpy>=2.0rc1`, workaround in https://github.com/NixOS/nixpkgs/pull/305152/commits/a00fc32b31cd002d8262fe4b15ddabd3e30580da

### Hydra jobset
https://hydra.lossy.network/jobset/nixpkgs/python-updates

### Eval reports
https://shells.darmstadt.ccc.de/~hexa/219.html

### CODE REVIEW

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/200.html

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/203.html

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/206.html

**Comment by drupol** [comment]
Can we upgrade to sphinx 7.3.7 in here so I can close https://github.com/NixOS/nixpkgs/pull/305335 ?

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/211.html

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/213.html

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/216.html

**Comment by mweinelt** [comment]
https://shells.darmstadt.ccc.de/~hexa/219.html

File: pkgs/development/python-modules/trio-asyncio/default.nix [hunk 1]
```diff
@@ -43,16 +44,18 @@ buildPythonPackage rec {
...
 
   nativeCheckInputs = [
     pytest-trio
     pytestCheckHook
   ];
 
-  disabledTestPaths = [
-    "tests/python" # tries to import internal API test.test_asyncio
+  disabledTests = [
+    #"test_trio_asyncio_cancel_direct"
```
Comment (mweinelt) [inline pkgs/development/python-modules/trio-asyncio/default.nix:58]: leftover?

**Comment by mweinelt** [comment]
@ofborg eval

**Comment by mweinelt** [comment]
Eval fails due to #288435.

**Pull request merged**